test/util/periods_unit.js

```diff
@@ -1141,14 +1149,163 @@ describe('PeriodCombiner', () => {
     expect(audio2.originalId).toBe('stream2,stream4');
   });
 
+  it('Matches streams with primary in common', async () => {
```

Comment:
This test simulates the stream that originally manifested the issue. It is a multi period steam where all periods have the same number of video and audio adaptation sets, and the audio adaptations both have roles, one `main` and one `description`.